src/components/Card.js

```diff
@@ -1,36 +1,62 @@
 export default class Card {
-  constructor(data, cardSelector, handleImageClick) {
+  constructor(
+    data,
+    cardSelector,
+    handleImageClick,
+    handleTrashButtonClick,
+    handleLikeButtonClick
+  ) {
     this._data = data;
     this._name = data.name;
     this._link = data.link;
+    this.id = data._id;
     this._cardSelector = cardSelector;
     this._handleImageClick = handleImageClick;
+    this._handleTrashButtonClick = handleTrashButtonClick;
+    this._handleLikeButtonClick = handleLikeButtonClick;
     this._template = document.querySelector(this._cardSelector).content;
     this._cardTemplate = this._template.querySelector(".card");
     this._cardElement = this._cardTemplate.cloneNode(true);
     this._likeButton = this._cardElement.querySelector(".card__like-button");
     this._deleteButton = this._cardElement.querySelector(
       ".card__delete-button"
     );
+    this._likeButtonStateActive = data.isLiked;
+    this._setLikeButtonState();
   }
 
   //like-button functionality
   _addLikeFunctionality() {
     this._likeButton.addEventListener("click", () => {
-      this._likeButton.classList.toggle("card__like-button_active");
+      this._handleLikeButtonClick(this, this._likeButtonStateActive);
+      //flip the state of the like button
+      this._likeButtonStateActive = !this._likeButtonStateActive;
+
+      this._setLikeButtonState();
```

Comment:
You need to **toggle likes** only in block `then` after a successful response from the server because if there is an error the user will not know about that and will see the toggled heart color.